### ISSUE TITLE
Interaction between TypeVarTuple and TypeVar leads to Unknown

### ISSUE THREAD

**Issue by jranieri-grammatech** [microsoft/pyright#6724]
**Describe the bug**
Having a TypeVarTuple followed by a TypeVar in a Callable leads to the TypeVar being left as Unknown.

**Code or Screenshots**
```python
from typing_extensions import (
    TypeVarTuple,
    TypeVar,
    Generic,
    Callable,
    Unpack,
    reveal_type
)

T = TypeVar("T")
TS = TypeVarTuple("TS")

class Dispatcher(Generic[Unpack[TS], T]):
    def set_result(self, result: T) -> None:
        ...

def make_dispatcher(x: Callable[[Unpack[TS], T], None]) -> Dispatcher[Unpack[TS], T]:
    return Dispatcher()

def callback(path: str, value: int) -> None:
    ...

d = make_dispatcher(callback)
d.set_result(None)  # <-- should issue an error about passing None when expecting an int but actually is accepted
reveal_type(d.set_result)  # <-- should say something like `(result: int) -> None` but actually says `(result: Unknown) -> None`
```

**Notes**
Moving the TypeVar to come before the Unpack results in pyright tracking the type correctly (albeit for a different parameter than I care about), so this seems to just be an issue with TypeVarTuple being too greedy.

**VS Code extension or command-line**
Tested against pyright 1.1.340. mypy 1.7.1 appears to handle this correctly.

**Comment by erictraut** [comment]
Thanks for the bug report. This is related to a complication that PEP 646 introduced to the type system (one that I unsuccessfully tried to get the authors of PEP 646 to omit from the spec). Prior to PEP 646, there was no way to define a function signature that included a `*args` parameter _followed by_ a positional-only parameter. Such a signature is considered a syntax error in a `def` statement. Or rather, any parameter after a `*args` is always assumed to be a keyword parameter. The annotation `Callable[[*Ts, T], None]` is effectively creating a signature with this previously-illegal combination. This breaks assumptions in a number of places within complex code in pyright. I thought I had found and fixed all of these places, but your code sample uncovers one that I hadn't previously found. I'll work on a fix.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.341, which I just published. It will also be included in a future release of pylance.
